src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.cs

```diff
@@ -341,11 +341,13 @@ protected virtual void ReportUnassignedOutParameter(ParameterSymbol parameter, S
             }
         }
 
-        protected static bool HasInitializer(FieldSymbol field)
+        protected static bool HasInitializer(Symbol field) => field switch
         {
-            return (field as SourceMemberFieldSymbol)?.HasInitializer == true ||
-                (field as SynthesizedBackingFieldSymbol)?.HasInitializer == true;
-        }
+            SourceMemberFieldSymbol f => f.HasInitializer,
+            SynthesizedBackingFieldSymbol f => f.HasInitializer,
+            SourceFieldLikeEventSymbol e => e.AssociatedEventField?.HasInitializer == true,
```

Comment:
>? [](start = 66, length = 1)

When does a SourceFieldLikeEventSymbol have a null AssociatedEventField?

Reply:
`extern`

Reply:
I'll add to the test